File: app/src/main/java/com/example/app/MainActivity.java
package com.example.app;

import static android.content.ContentValues.TAG;

import androidx.appcompat.app.AppCompatActivity;
import com.chaquo.python.*;
import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import com.chaquo.python.android.AndroidPlatform;
import com.example.app.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    // Used to load the 'app' library on application startup.
    static {
        System.loadLibrary("app");
    }

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        if (! Python.isStarted()) {
            Python.start(new AndroidPlatform(this));
        }

        // Função Java

        long startJv = System.currentTimeMillis();
        PiBenchGameJava.run(10000);
        long endJv = System.currentTimeMillis();

        long finalJv = endJv - startJv;

        // Função Kotlin

        long startKt = System.currentTimeMillis();
        PiBenchGameKtKt.run(10000);
        long endKt = System.currentTimeMillis();

        long finalKt = endKt - startKt;

        // Função Golang

        long startGo = System.currentTimeMillis();
        // Função
        long endGo = System.currentTimeMillis();

        long finalGo = endGo - startGo;

        // Função Python

        Python py = Python.getInstance();
        PyObject obj = py.getModule("PyDigitis");

        long startPy = System.currentTimeMillis();
        PyObject result = obj.callAttr("pidigits", 10000);
        long endPy = System.currentTimeMillis();

        long finalPy = endPy - startPy;

        // Função Rust

        long startRs = System.currentTimeMillis();
        // Função
        long endRs = System.currentTimeMillis();

        long finalRs = endRs - startRs;

        // Função CPP

        long startCpp = System.currentTimeMillis();
        int valor = cppidigits(10000);
        long endCpp = System.currentTimeMillis();

        long finalCpp = endCpp - startCpp;

        // Função C

        long startC = System.currentTimeMillis();
        int valor2 = cpidigits(10000);
        long endC = System.currentTimeMillis();

        long finalC = endC - startC;

    }

    /**
     * A native method that is implemented by the 'app' native library,
     * which is packaged with this application.
     */

    public native int cppidigits(int num);
    public native int cpidigits(int num);
}

File: app/src/main/java/com/example/myappbench/MainActivity.kt
package com.example.myappbench

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.chaquo.python.Python
import com.chaquo.python.android.AndroidPlatform
import com.example.myappbench.FannkuchBenchGameJava.main
import com.example.myappbench.MeasureMemory.MeasureMemory


class MainActivity : AppCompatActivity() {

    companion object {
        init {
            System.loadLibrary("myappbench")
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Python Start

        /*if (! Python.isStarted()) {
            Python.start(AndroidPlatform(this))
        }

        val py = Python.getInstance()
        val pyobj = py.getModule("PyBinaryTrees")
        val startPy = System.currentTimeMillis()
        pyobj.callAttr("binary_trees", 21)
        val endPy = System.currentTimeMillis()
        val finalPy = endPy - startPy
        Log.i("Tempo de execucao Python Binary Trees:", finalPy.toString())

        // Python End*/

        // Java Start

        val startJv = System.currentTimeMillis()
        FannkuchBenchGameJava.main(12)
        //TreeBenchGameJava.run(21)
        //PiBenchGameJava.pidigitsjv(10000)
        //BodyBenchGameJava.run(50000000)
        val endJv = System.currentTimeMillis()
        val finalJv = endJv - startJv
        Log.i("Java Fannkuch:", finalJv.toString())

        // Java End

        // Kotlin Start

        val startKt = System.currentTimeMillis()
        //PiBenchGameKt().run(10000)
        //TreeBenchGameKt().run(21)
        FannkuchBenchGameKt().main(12)
        //BodyBenchGameKt().run(50000000)
        val endKt = System.currentTimeMillis()
        val finalKt = endKt - startKt
        Log.i("Kotlin fannkuch:", finalKt.toString())

        // Kotlin End

        // C Start

        val startC = System.currentTimeMillis()
        cFannkuch(12)
        // cNbody(50000000)
        // cBinaryTrees(21)
        // cpidigits(10000);
        val endC = System.currentTimeMillis()
        val finalC = endC - startC
        Log.i("C fannkuch:", finalC.toString())

        // C End

        // C++ Start

        val startCpp = System.currentTimeMillis();
        cppFannkuch(12);
        // cppNbody(50000000);
        // cppidigits(10000);
        // cppBinaryTrees(21);
        val endCpp = System.currentTimeMillis();
        val finalCpp = endCpp - startCpp;
        Log.i("Cpp Fannkuch: ", finalCpp.toString());

        // C++ End

    }

    external fun cppidigits(num: Int)
    external fun cpidigits(num: Int)

    external fun cppFannkuch(num: Int)
    external fun cFannkuch(num: Int)

    external fun cppBinaryTrees(num: Int)
    external fun cBinaryTrees(num: Int)

    external fun cppNbody(num: Int)
    external fun cNbody(num: Int)

}